source/localizable/tutorial/subroutes.md

```diff
@@ -0,0 +1,398 @@
+  image: DS.attr(),
+  bedrooms: DS.attr(),
+  slug: DS.attr(),
+  description: DS.attr()
+});
+```
+
+# Generating a Show Route
+
+Now that our API is ready to return single records, we can generate our `show` sub-route.
+Much like generating our `rentals` route, we will use `ember g` to create a nested route.
+
+```shell
+ember g route rentals/show
+```
+
+You will see output like this:
+
+```shell
+installing route
+  create app/routes/rentals/show.js
+  create app/templates/rentals/show.hbs
+updating router
+  add route rentals/show
+installing route-test
+  create tests/unit/routes/rentals/show-test.js
+```
+
+Let's start by looking at the changes to our Router (`app/router.js`).
+
+```app/router.js
+Router.map(function() {
+  this.route('about');
+  this.route('contact');
+  this.route('rentals', function() {
+    this.route('show');
+  });
+});
+```
+
+You will notice that `this.route('show')` is nested within our `rentals` route.
+This tells Ember that it is a sub-route and must be accessed through `localhost:4200/rentals/show`.
+
+In order for us to tell the application which rental we want to access, we need to pass the `slug` to our `show` route.
+In addition, we will want to simplify the URL for the user so that they can access the information for a specific rental by browsing to `localhost:4200/rentals/slug-for-rental`.
+To do this, we modify our router.
+
+```app/router.js
+Router.map(function() {
+  this.route('about');
+  this.route('contact');
+  this.route('rentals', function() {
+    this.route('show', { path: '/:slug' });
+  });
+});
+```
+
+We have modified the `path` for the `show` route.
+It no longer defaults to `/rentals/show` but instead `/rentals/:slug`.
+This means we can now access the `slug` on the route itself.
+
+# Querying by Slug
+
+Next, we will want to edit our `show` route to return the specific rental we want.
+
+```app/routes/rentals/show.js
+export default Ember.Route.extend({
+  model(params) {
+    return this.store.queryRecord('rental', { slug: params.slug });
+  }
+});
+```
+
+Since we added `:slug` to the `show` path in our router, we can now access `slug` through the `params` in our `model` hook.
+When we call `this.store.queryRecord('rental', { slug: params.slug })`, Ember Data will make a GET request to `/rentals?slug=our-slug`.
+You can read more about Ember Data in the [Models section](../../models/).
+
+To ensure this, we should write a unit test that confirms `queryRecord` is called with the correct parameters.
+Let's edit the unit test that Ember created for us when we generated our `show` route.
+
+This means that we can access a specific rental as `model` on our template.
+
+```tests/unit/routes/rentals/show-test.js
+moduleFor('route:rentals/show', 'Unit | Route | rentals/show', {});
+
+test('should query for selected rental by slug', function(assert) {
+  let store = {
+    queryRecord(path, options) {
+      assert.equal(path, 'rental', 'queryRecord calls rental path on API');
```

Comment:
Instead of asserting in your stub, just return a resolved promise simulating the return value and assert on the result of the `model` function.